### PR TITLE
Add generate signed token endpoint

### PR DESCRIPTION
**Pull request by lucperkins**
I was hoping to add this operation (generating a signed ID token) to vaultrs: https://developer.hashicorp.com/vault/api-docs/secret/identity/tokens#generate-a-signed-id-token.

### CODE REVIEW

**Comment by Haennetz** [comment]
I think it would be a good idea to use a new module structure like this https://github.com/jmgilman/vaultrs/tree/master/src/api/auth for the identity secret engine.

**Comment by Haennetz** [comment]
Please rebase your branch with the current master we merged a pull request that also added the identity module

**Comment by lucperkins** [comment]
@Haennetz Done

File: src/api/identity/identity_tokens/requests.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+use super::responses::GenerateSignedIdTokenResponse;
+use rustify_derive::Endpoint;
+
+/// ## Generate a signed token ID
+/// This endpoint generates a signed ID (OIDC) token.
+///
+/// * Path: /identity/oidc/token/{self.role}
+/// * Method: GET
+/// * Response: [GenerateSignedIdTokenResponse]
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/tokens#generate-a-signed-id-token
```
Comment (Haennetz) [inline src/api/identity/identity_tokens/requests.rs:10]:
```suggestion
/// * Reference: <https://developer.hashicorp.com/vault/api-docs/secret/identity/tokens#generate-a-signed-id-token>
```

File: src/api/identity/identity_tokens/responses.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+use serde::{Deserialize, Serialize};
+
+/// Response from executing
+/// [GenerateSignedIdTokenRequest][crate::api::identity::requests::GenerateSignedIdTokenRequest]
```
Comment (Haennetz) [inline src/api/identity/identity_tokens/responses.rs:4]:
```suggestion
/// [GenerateSignedIdTokenRequest][crate::api::identity::identity_tokens::requests::GenerateSignedIdTokenRequest]
```

File: nix/automation/devshells.nix [hunk 3]
```diff
@@ -25,6 +25,7 @@
       gcc
       rustToolchain
       pkg-config
+      libiconv
```
[the remark this answers is not in the document]
Comment (lucperkins) [reply nix/automation/devshells.nix]: It's required for macOS, yes. In a standard flake I'd be able to make this specific to `*-darwin` but I don't know how to in this framework. I'll leave it out.

File: src/api/auth/identity.rs
Comment (Haennetz) [inline src/api/auth/identity.rs:1]: I think this is a leftover from an earlier commit.

**Comment by lucperkins** [comment]
@jmgilman Please let me know if there's anything else required to get this merged.

**Comment by Haennetz** [comment]
Thanks for the PR.  It would be nice if we can cover the functionality in a short test.

**Comment by lucperkins** [comment]
@Haennetz The issue here is that covering this properly in a test would entail adding some other endpoints as well. Specifically, it requires an already created [role](https://developer.hashicorp.com/vault/api-docs/secret/identity/tokens#name-7) and as far as I can tell that isn't covered in this integration yet.

**Comment by lucperkins** [comment]
I could conceivably add that endpoint but I was hoping to keep this PR as single purpose as possible

**Comment by lucperkins** [comment]
@Haennetz Okay, I've added a test case here with the required second endpoint.

**Comment by lucperkins** [comment]
@Haennetz Okay, the test error has been addressed

**Comment by Haennetz** [comment]
@lucperkins the test failes because it tries to add a new role with the name of a named key that dosn't exist.

**Comment by lucperkins** [comment]
@Haennetz Yeah, I just don't really understand why a role isn't created when I do this:

```rust
let role_name = "my-role";
let key = "named-key-001";
create_or_update_role(&client, role_name, key).await;
test_generate_id_token_by_role_name(&client, role_name).await;
```

This is why I initially didn't add a test for generating a token ID. I just don't really understand the client well enough to set up token generation in the first place.

**Comment by lucperkins** [comment]
@Haennetz I'm going to close this. I unfortunately don't really have the bandwidth to contribute a test here. I'm going to use a fork instead.

**Comment by Haennetz** [comment]
The issue is that the key you try to use in the role doesn't exist in vault. If find the time I will add all api endpoints for the OIDC identity provider to it. With that we are able to create the required key to create a role

**Comment by lucperkins** [comment]
@Haennetz Right, and so I'd need to add yet another endpoint for that. I'd be happy to re-open this PR without a test. I do think that that would be beneficial for users.

**Comment by Haennetz** [comment]
If it is okay for, you I will include your changes without the test in a new feature branch,as a squashed commit.  Then I will add the tests and more endpoints to include the whole OIDC identity provide api endpoints

**Comment by lucperkins** [comment]
@Haennetz Sounds great! Thanks a lot for your work on this great project.

**Comment by Haennetz** [comment]
And you for participation to it

**Pull request closed without merging**